packages/@angular/cli/utilities/check-package-manager.ts

```diff
@@ -0,0 +1,42 @@
+import * as chalk from 'chalk';
+import {exec} from 'child_process';
+
+import {CliConfig} from '../models/config';
+
+const Promise = require('../ember-cli/lib/ext/promise');
+const config = CliConfig.fromProject();
+
+const execPromise = Promise.denodeify(exec);
+
+let packageManager = 'npm';
+if (config && config.get('packageManager')) {
+  packageManager = config.get('packageManager');
+}
+
+export function checkYarnOrCNPM() {
+  return Promise
+      .all([checkYarn(), checkCNPM()])
+      .then((data: Array<boolean>) => {
+        if (packageManager === 'npm') {
```

Comment:
So you're checking both yarn and cnpm EVEN if the user asked for a specific one? That's a lot of unnecessary work (and will slow down the bootstrapping time significantly).